src/plugins/intel_npu/src/compiler_adapter/src/vcl_api.cpp

```diff
@@ -186,6 +186,27 @@ struct vcl_allocator_vector : vcl_allocator2_t {
     std::vector<uint8_t> m_vec;
 };
 
+struct vcl_allocator_vector_2 : vcl_allocator2_t {
+    vcl_allocator_vector_2() : vcl_allocator2_t{vector_allocate, vector_deallocate} {}
+
+    static uint8_t* vector_allocate(vcl_allocator2_t* allocator, size_t size) {
+        vcl_allocator_vector_2* vecAllocator = static_cast<vcl_allocator_vector_2*>(allocator);
+        auto newVec = std::make_shared<std::vector<uint8_t>>();
+        newVec->resize(size);
+        uint8_t* ptr = newVec->data();
+        vecAllocator->m_vector.emplace_back(std::make_pair(ptr, newVec));
+        return ptr;
+    }
+
+    static void vector_deallocate(vcl_allocator2_t* allocator, uint8_t* ptr) {
+        vcl_allocator_vector_2* vecAllocator = static_cast<vcl_allocator_vector_2*>(allocator);
+        vecAllocator->m_vector.clear();
+        vecAllocator->m_vector.shrink_to_fit();
+    }
+
+    std::vector<std::pair<uint8_t*, std::shared_ptr<std::vector<uint8_t>>>> m_vector;
```

Comment:
```suggestion
    std::vector<std::shared_ptr<std::vector<uint8_t>>> m_vector;
```
I'm wondering: do we need to store this pointer separately? You can always extract it using vector->data() if you need it, right? I also haven't noticed any place where the first value is used, but I also admit I didn't look very carefully after it.

Reply:
Thx! Changed it to `std::vector<std::shared_ptr<std::vector<uint8_t>>> m_vector;` as the pointer is not needed.